figure_notebooks/figure4.md

```diff
@@ -0,0 +1,997 @@
+plot_df = subset(top_diff_MC, tot_citations > MIN_ART | tot_mentions > MIN_ART)
+plot_df$idx = paste(plot_df$year, plot_df$country)
+plot_df$filtered = "filter"
+keep_idx = paste(top_diff_MC_filt$year, top_diff_MC_filt$country)
+plot_df$filtered[plot_df$idx %in% keep_idx] = "keep"
+
+# show the spread of the difference mentions and citations
+c_vs_m_filter_gg = ggplot(plot_df, aes(x=as.numeric(year), 
+                        y=as.numeric(M_C),
+                        color = filtered)) +
+    geom_point() + theme_bw() + 
+    xlab("Corpus") + ylab("Mention % - Citation % for each country+year") +
+    ggtitle("Diff. btw mentions and citations for each country+year (1 point is a country)") + 
+    scale_fill_brewer(palette="Set2")
+ggsave(file.path(proj_dir, "/figure_notebooks/tmp_files/fig4_tmp/c_vs_m_filter_gg.pdf"),
+       c_vs_m_filter_gg, width = 10, height = 5, units = "in", device = "pdf")
+
+
+## make the overall heatmap
+make_heatmap_res <- function(in_df, value_col){
+    plot_matr_MC = reshape2::dcast(in_df, 
+                             country ~ year, 
+                             value.var=value_col)
+    row.names(plot_matr_MC) = plot_matr_MC$country
+    plot_matr_MC = plot_matr_MC[,-1]
+    plot_matr_MC[is.na(plot_matr_MC)] = 0
+    
+    max_val = max(abs(plot_matr_MC), na.rm = T)
+    breaks = c(seq(-1*max_val, max_val, by = 0.01))
+    color_pmap <- colorRampPalette(c("pink", "white", "green"))(length(breaks))
+
+    if(max_val > 1){
+        breaks = c(seq(1, max_val, by = 1))
+        color_pmap <- colorRampPalette(c("white", "green"))(length(breaks))
+
+    }
+
+    res = list(plot_matr = plot_matr_MC,
+               color_pmap = color_pmap,
+               breaks = breaks)
+    return(res)
+}
+
+# plot the Top proportion differences
+countries_considered = union(class_m_mentions$est_country,
+                             class_c_mentions$est_country)
+heatmap_df = subset(top_diff_MC, country %in% countries_considered)
+res_MC = make_heatmap_res(heatmap_df, value_col="M_C")
+
+# make annotation of the median difference
+heatmap_median_df = heatmap_df %>% 
+                group_by(country) %>% 
+                summarise(median_M_C = median(M_C)) 
+annot_df = data.frame(median_M_C = heatmap_median_df$median_M_C)
+row.names(annot_df) = heatmap_median_df$country
+annot_df$class = "Cite"
+annot_df$class[annot_df$median_M_C > 0] = "Mention"
+annot_df$class = factor(annot_df$class, levels = c("Mention", "Cite"))
+
+annot_df = annot_df[order(annot_df$median_M_C),]
+
+
+# now make the heatmap
+res_MC$plot_matr = res_MC$plot_matr[row.names(annot_df),]
+full_heatmap = pheatmap(res_MC$plot_matr, cluster_rows = F, 
+         cluster_cols = F, display_numbers = T, 
+         main = "Top (Mention - Citation) Proportions",
+         color = res_MC$color_pmap, breaks = res_MC$breaks,
+         number_color = "black", annotation_row = annot_df)
+```
+
+![](figure4_files/figure-markdown_github/make_heatmap_gg-1.png)
```

Comment:
Heat map looks a bit cramped in this part, but doesn't really matter as it looks great in the main figure. Nice!!